bigquery/google/cloud/bigquery/_pandas_helpers.py

```diff
@@ -245,6 +267,21 @@ def dataframe_to_bq_schema(dataframe, bq_schema):
     return tuple(bq_schema_out)
 
 
+def _column_or_index(dataframe, name):
+    """Return a column or index as a pandas series."""
+    if name in dataframe.columns:
+        return dataframe[name]
+
+    if isinstance(dataframe.index, pandas.MultiIndex):
+        if name in dataframe.index.names:
+            return dataframe.index.get_level_values(name)
+    else:
+        if name == dataframe.index.name:
+            return dataframe.index.to_series()
+
+    raise ValueError("column or index '{}' not found.".format(name))
```

Comment:
Formatting the error message will fail under Python 2 if index name contains non-ASCII characters, and a `UnicodeEncodeError` is probably not all that informative.

It can happen if a non-ASCII field name ends up in `bq_schema` passed to the  `dataframe_to_arrow()` helper, although that's probably unlikely, given that non-ASCII names are not allowed in table column names.